### ISSUE TITLE
failed to select a version for the requirement `anchor-lang = "^0.22.0"

### ISSUE THREAD

**Issue by anoushk1234** [cronos-so/cronos#21]
```
failed to select a version for the requirement `anchor-lang = "^0.22.0"`
candidate versions found which didn't match: 0.24.2
location searched: crates.io index
required by package `anchor-client v0.22.0`
    ... which satisfies dependency `anchor-client = "=0.22.0"` of package `cronos-sdk v0.1.8`
    ... which satisfies dependency `cronos-sdk = "^0.1.8"` of package `sol-ipl v0.1.0 
```

**Comment by anoushk1234** [comment]
https://github.com/cronos-so/anchor/pull/6 this should fix it

**Comment by nickgarfield** [comment]
Hey @anoushk1234! Loving these PRs. 

Our main devops challenge here is that we need to build Cronos for testnet (right now 1.10.*) with all the latest updates and interfaces, but Anchor doesn't update their Solana dependencies until those versions are live and stable on mainnet-beta. 

To achieve this, we're maintaining a fork of Anchor (https://github.com/cronos-so/anchor) that updates all the `solana-*` dependencies to the latest beta channel release of Solana. We cut branches for those versions (eg [v0.24.2-solana.1.10.8](https://github.com/cronos-so/anchor/tree/v0.24.2-solana.1.10.8)) and reference those branches from this repo. 

I'm not running into this build error on main at the moment, but if you're able to show the steps that lead up to it, I can reproduce it and offer suggestions for how to work around it.

**Comment by nickgarfield** [comment]
Basically, in this repo we only want to use:

```
anchor-lang = { git = "https://github.com/cronos-so/anchor", branch = "v0.24.2-solana.1.10.8" }
``` 

and will only update the Anchor dependencies to a stable release version once Anchor provides a build that supports our required Solana version (ie once it's stable on mainnet).

**Comment by anoushk1234** [comment]
> Hey @anoushk1234! Loving these PRs.
> 
> Our main devops challenge here is that we need to build Cronos for testnet (right now 1.10.*) with all the latest updates and interfaces, but Anchor doesn't update their Solana dependencies until those versions are live and stable on mainnet-beta.
> 
> To achieve this, we're maintaining a fork of Anchor (https://github.com/cronos-so/anchor) that updates all the `solana-*` dependencies to the latest beta channel release of Solana. We cut branches for those versions (eg [v0.24.2-solana.1.10.8](https://github.com/cronos-so/anchor/tree/v0.24.2-solana.1.10.8)) and reference those branches from this repo.
> 
> I'm not running into this build error on main at the moment, but if you're able to show the steps that lead up to it, I can reproduce it and offer suggestions for how to work around it.

I was running into build errors when i import cronos and anchor both from crates.io instead of git but ill try to import your fork of anchor to see if it works
if you want to reproduce my issue just import thes in cargo toml
```toml
[dependencies]
anchor-lang = "0.24.2"
anchor-spl ="0.24.2"
cronos-sdk = "0.0.1"
```

**Comment by anoushk1234** [comment]
Hey @nickgarfield update on the dependency issue, after importing the cronos anchor fork im still getting this error
```toml
[dependencies]
anchor-lang = { git = "https://github.com/cronos-so/anchor", branch = "v0.24.2-solana.1.10.8" }
anchor-spl = { features = ["mint", "token","associated_token"], git = "https://github.com/cronos-so/anchor", branch = "v0.24.2-solana.1.10.8" }
cronos-sdk = "0.1.8"
```
```
failed to select a version for the requirement `anchor-lang = "^0.22.0"`
candidate versions found which didn't match: 0.24.2
location searched: crates.io index
required by package `anchor-client v0.22.0`
    ... which satisfies dependency `anchor-client = "=0.22.0"` of package `cronos-sdk v0.1.8`
    ... which satisfies dependency `cronos-sdk = "^0.1.8"` of package `sol-ipl v0.1.0 (/Users/anoushkkharangate/Documents/solana-programs/sol-ipl/programs/sol-ipl)
```

**Comment by nickgarfield** [comment]
Going to leave this open until we publish v0.2 crates. 

Since Anchor versions <0.24.2 have been yanked, there isn't currently a way to compile v0.1 builds. The interface changes between v0.1 and v0.2 is so large, it's really not worth publishing a fix for v0.1 branches. 

v0.2 will land soon and Anchor programs will compile with the [cronos-so/anchor](https://github.com/cronos-so/anchor) repo.

**Comment by anoushk1234** [comment]
> Going to leave this open until we publish v0.2 crates.
> 
> Since Anchor versions <0.24.2 have been yanked, there isn't currently a way to compile v0.1 builds. The interface changes between v0.1 and v0.2 are so large, it's also not really worth publishing a fix for v0.1 branches.
> 
> v0.2 will land soon and Anchor programs will compile with the [cronos-so/anchor](https://github.com/cronos-so/anchor) repo – unblocking devs to build Cronos programs for devnet/testnet.

got it, will wait till then thanks!

**Comment by kai-wen-zhang** [comment]
looks like i encountered a similar error 
```
error: failed to compile `cronos-cli v0.1.8`, intermediate artifacts can be found at `/var/folders/2g/p4h7pp6n4ssgcc9p1470kll00000gn/T/cargo-installhO6mUe`

Caused by:
  failed to select a version for the requirement `anchor-lang = "=0.22.0"`
  candidate versions found which didn't match: 0.25.0, 0.24.2
  location searched: crates.io index
  required by package `cronos-cli v0.1.8`
```
